.github/workflows/tests.yml

```
@@ -256,6 +256,9 @@ jobs:
 
         with_backoff balena device ${{ steps.register-test-device.outputs.balena_device_uuid }}
 
+      env:
+        COMMIT: ${{ github.event.pull_request.head.sha || github.event.head_commit.id || github.event.pull_request.head.ref }}
```

```suggestion
        COMMIT: ${{ github.event.pull_request.head.sha  }}
```
Just use the head sha if possible.